subprojects/base-services/src/main/java/org/gradle/internal/hash/FileHasher.java

```diff
@@ -24,7 +24,7 @@
 
 public interface FileHasher {
     /**
-     * Returns the hash of the given input stream.
+     * Returns the hash of the given input stream. This method buffers reads on the stream.
```

Comment:
We could add a follow up issue. For this one, I think it's not a good idea to add this information. It would imply that we only use buffers in places where the Javadoc says so, which is not true. Instead, currently you must look at the code you call.